tests/unit/api/objectPut.js

```diff
@@ -230,3 +238,102 @@ describe('objectPut API', () => {
         });
     });
 });
+
+describe('objectPut API with versioning', () => {
+    beforeEach(() => {
+        cleanup();
+    });
+
+    const objData = ['foo0', 'foo1', 'foo2'].map(str =>
+        Buffer.from(str, 'utf8'));
+    const testPutObjectRequests = objData.map(data => versioningTestUtils
+        .createPutObjectRequest(bucketName, objectName, data));
+
+    it('should delete latest version when creating new null version ' +
+    'if latest version is null version', done => {
+        async.series([
+            callback => bucketPut(authInfo, testPutBucketRequest, log,
+                callback),
+            // putting null version by putting obj before versioning configured
+            callback => objectPut(authInfo, testPutObjectRequests[0], undefined,
+                log, err => {
+                    versioningTestUtils.assertDataStoreValues(ds,
+                        objData.slice(0, 1));
+                    callback(err);
+                }),
+            callback => bucketPutVersioning(authInfo, suspendVersioningRequest,
+                log, callback),
+            // creating new null version by putting obj after ver suspended
+            callback => objectPut(authInfo, testPutObjectRequests[1],
+                undefined, log, err => {
+                    // wait until next tick since mem backend executes
+                    // deletes in the next tick
+                    process.nextTick(() => {
+                        // old null version should be deleted
+                        objData[0] = undefined;
```

Comment:
Fair, I think I was slicing the arrays at first because I initially was testing with more values than just three. I updated to list the values each time instead of slicing.

> (possibly modifying assertDataStoreValues() to take data strings as input)
do you providing strings to assertDataStoreValues() instead of the data buffers?